src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1426,10 +1426,81 @@ public static unsafe nuint NarrowUtf16ToAscii(char* pUtf16Buffer, byte* pAsciiBu
             goto Finish;
         }
 
-        private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pAsciiBuffer, nuint elementCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool VectorContainsNonAsciiChar(Vector128<ushort> utf16Vector, Vector128<short> asciiMaskForTestZ, Vector128<ushort> asciiMaskForAddSaturate)
```

Comment:
I am little uncomfortable with having a same method that takes parameters that are not relevant to AdvSimd. But on the other hand, if we have separate methods like `VectorContainsNonAsciiChar_SSE2` and `VectorContainsNonAsciiChar_AdvSimd`, we will have to add checks for `AdvSimd.IsSupported`, etc. at their usage. Not that they are expensive (they are constant prop and unsupported code should be removed), but it makes code harder to read.

Reply:
Agreed. Previously, the method only took utf16Vector as an argument. The other arguments (`asciiMaskForTestZ` and `ascciiMaskForAddSaturate`) were allocated in this method itself with a hope that inlining + CSE in would address this. However, it caused slowdown, especially, on x86 :( . Moving these allocations to the parent method reclaimed the lost performance on x86.

Reply:
> hope that inlining + CSE in would address this

So was it not taken care of by CSE?

Reply:
With the latest HEAD, I can confirm the CSE has kicked in and there is no duplicate allocation of constants in registers. Not sure if something changed recently. Redoing benchmarks. If there is no regression on x86, will switch back to the version that allocates constants/masks in the `VectorContainsNonAsciiChar` method.

Reply:
Updated the `VectorContainsNonAsciiChar` to take fewer arguments. JIT inlines the method and for both the same instructions are emitted to check for a non-ASCII char.

```
    vmovdqu  xmm0, xmmword ptr [rdi]
    vptest   xmm0, xmmword ptr [reloc @RWD00]
    jne      ...
    ...
    RWD00   dq      FF80FF80FF80FF80h, FF80FF80FF80FF80h
```

Reply:
Could you please share the complete disassembly of the method?